_includes/specials/policy/en/07-network-resources.md

```diff
@@ -0,0 +1,92 @@
+represent a major portion of all memory accesses of a node. Bitcoin Core
+already uses a [manually optimized data structure for the UTXO cache][pooled
+resource], but the size of the UTXO set determines how much of it cannot fit in
+a node’s cache. A larger UTXO set means more cache misses which slows down
+block validation, IBD, and transaction validation speed. The dust limit is an
+example of a policy that restricts the creation of UTXOs, specifically curbing
+UTXOs that might never get spent because their amount falls short of the cost
+for spending them. Even so, [“dust storms” with thousands of transactions
+occurred as recently as 2020][lopp storms].
+
+When it became popular to use bare multisig outputs to publish data onto the
+blockchain, the definition of standard transactions was amended to permit a
+single OP_RETURN output as an alternative. People realized that it would be
+impossible to prevent users from publishing data on the blockchain, but at
+least such data would not need to live in the UTXO set forever when published
+in outputs that could never be spent. Bitcoin Core 0.13.0 introduced a start-up
+option `-permitbaremultisig` that users may toggle to reject unconfirmed
+transactions with bare multisig outputs.
+
+While the consensus rules allow output scripts to be freeform, only a few
+well-understood patterns are relayed by Bitcoin Core nodes. This makes it
+easier to reason about many concerns in the network, including validation costs
+and protocol upgrade mechanisms. For example, an input script that contains
+op-codes, a P2SH input with more than 15 signatures, or a P2WSH input whose
+witness stack has more than 100 items each would make a transaction
+non-standard. (Check out this [policy overview][instagibbs policy zoo] for more
+examples of policies and their motivations.)
+
+Finally, the Bitcoin protocol is a living software project that will need to
+keep evolving to address future challenges and user needs. To that end, there
+are a number of upgrade hooks deliberately left consensus valid but unused,
+such as the annex, taproot leaf versions, witness versions, OP_SUCCESS, and a
+number of no-op opcodes. However, just like attacks are hindered by the lack of
+central points of failure, network-wide software upgrades involve a coordinated
+effort between tens of thousands of independent node operators. Nodes will not
+relay transactions that make use of any reserved upgrade hooks until their
+meaning has been defined. This discouragement is meant to dissuade applications
+from independently creating conflicting standards, which would make it
+impossible to adopt one application’s standard into consensus without
+invalidating another’s. Also, when a consensus change does happen, nodes that
+do not upgrade immediately—and thus do not know the new consensus rules—cannot
+be “tricked” into accepting a now-invalid transaction into their mempools. The
```

Comment:
> accepting a now-invalid transaction into their mempools

This is an excellent point, but I wonder if casual readers will intuit the ultimate consequences of this behavior, e.g. that ordinary mempools may be flooded with high-feerate now-invalid transactions to the point where they won't see ordinary valid transactions or be able to use BIP152, or that miners using old node software might produce invalid blocks, costing them huge amounts of money.

I think it might be more compelling if you add some text about that, but I also think it's fine as-is.

Reply:
saw this too late gah :see_no_evil: thanks, that would've been a great addition